clients/gateway.go

```diff
@@ -87,32 +86,35 @@ func (c *GatewayClient) buildQueryString(endpoint string, args map[string]string
 func (c *GatewayClient) get(ctx context.Context, queryUrl string) ([]byte, error) {
 	var res *http.Response
 	var err error
-	wait := c.minWait
+	wait := time.Duration(0)
 	for i := 0; i <= c.maxRetries; i++ {
-		req, err := http.NewRequestWithContext(ctx, "GET", queryUrl, nil)
-		if err != nil {
-			return nil, err
-		}
-
-		res, err = c.client.Do(req)
-		if res != nil && res.StatusCode == http.StatusOK {
-			break
-		} else if err != nil {
-			return nil, err
-		}
-
-		wait = c.backoff(wait)
-		if wait > c.maxWait {
-			wait = c.maxWait
+		select {
+		case <-ctx.Done():
+			return nil, errors.New("canceled")
+		case <-time.After(wait):
+			var req *http.Request
+			req, err = http.NewRequestWithContext(ctx, "GET", queryUrl, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			res, err = c.client.Do(req)
+			if err == nil && res != nil && res.StatusCode == http.StatusOK {
+				return io.ReadAll(res.Body)
+			} else if res.StatusCode != http.StatusOK {
+				err = errors.New(res.Status)
+			}
+
+			if wait < c.minWait {
+				wait = c.minWait
+			}
+			wait = c.backoff(wait)
+			if wait > c.maxWait {
+				wait = c.maxWait
+			}
 		}
 	}
-	if err != nil {
-		return nil, err
-	} else if res.StatusCode != http.StatusOK {
-		return nil, errors.New(res.Status)
-	}
-
-	return io.ReadAll(res.Body)
+	return nil, err
```

Comment:
Will this return a nil error after `maxRetries`? If so, maybe we can create a new error?

Reply:
It just returns the last error `Do` returned. am I missing something?

Reply:
I believe it will the error on line 105

Reply:
or what L105 sets it to